From ffecd70ac21e2fdc5532e140bf5799f86c358f20 Mon Sep 17 00:00:00 2001
From: Patrick Huesmann <patrick.huesmann@desy.de>
Date: Tue, 19 Dec 2023 10:44:23 +0100
Subject: [PATCH 1/4] Add support for MMC mailbox

---
 drivers/misc/i2c_eeprom.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/misc/i2c_eeprom.c b/drivers/misc/i2c_eeprom.c
index 6c0459dc55..7f3002e34d 100644
--- a/drivers/misc/i2c_eeprom.c
+++ b/drivers/misc/i2c_eeprom.c
@@ -269,6 +269,7 @@ static const struct udevice_id i2c_eeprom_std_ids[] = {
 	{ .compatible = "atmel,24c128", (ulong)&atmel24c128_data },
 	{ .compatible = "atmel,24c256", (ulong)&atmel24c256_data },
 	{ .compatible = "atmel,24c512", (ulong)&atmel24c512_data },
+	{ .compatible = "desy,mmcmailbox", (ulong)&atmel24c32_data },
 	{ }
 };
 
-- 
2.43.0

