From 54d6cd9a47173af391844266e7677574299fc044 Mon Sep 17 00:00:00 2001
From: Patrick Huesmann <patrick.huesmann@desy.de>
Date: Mon, 14 Mar 2022 12:22:56 +0100
Subject: [PATCH] Add custom default env settings

---
 include/configs/xilinx_zynqmp.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/configs/xilinx_zynqmp.h b/include/configs/xilinx_zynqmp.h
index 5891d26385..22871a1cfe 100644
--- a/include/configs/xilinx_zynqmp.h
+++ b/include/configs/xilinx_zynqmp.h
@@ -283,6 +283,8 @@
 
 #include <config_distro_bootcmd.h>
 
+#include "custom_default_env.h"
+
 /* Initial environment variables */
 #ifndef CONFIG_EXTRA_ENV_SETTINGS
 #define CONFIG_EXTRA_ENV_SETTINGS \
diff --git a/include/configs/zynq-common.h b/include/configs/zynq-common.h
index d0614d8da4..0e4cd7c4e2 100644
--- a/include/configs/zynq-common.h
+++ b/include/configs/zynq-common.h
@@ -195,6 +195,8 @@
 #include <config_distro_bootcmd.h>
 #endif /* CONFIG_SPL_BUILD */
 
+#include "custom_default_env.h"
+
 /* Default environment */
 #ifndef CONFIG_EXTRA_ENV_SETTINGS
 #define CONFIG_EXTRA_ENV_SETTINGS	\
-- 
2.25.1

