From 6a80e922a68c6863a15558029a97612216db9929 Mon Sep 17 00:00:00 2001
From: Patrick Huesmann <patrick.huesmann@desy.de>
Date: Fri, 23 Feb 2024 14:56:35 +0100
Subject: [PATCH] Disable PHY init

---
 drivers/net/zynq_gem.c | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/drivers/net/zynq_gem.c b/drivers/net/zynq_gem.c
index 1cf4477b4f..c9f5496505 100644
--- a/drivers/net/zynq_gem.c
+++ b/drivers/net/zynq_gem.c
@@ -745,9 +745,13 @@ static int zynq_gem_probe(struct udevice *dev)
 	if (ret)
 		goto err2;
 
-	ret = zynq_phy_init(dev);
-	if (ret)
-		goto err3;
+	// On the DAMC-MOTCTRL, eth0 is directly wired to the backplane, without a PHY
+	// in between Initialize the GEM interface only, w/o trying to initialize a
+	// non-existing PHY
+	//
+	// ret = zynq_phy_init(dev);
+	// if (ret)
+	//	goto err3;
 
 	return ret;
 
-- 
2.43.2

